src/main/java/org/opensearch/neuralsearch/query/NeuralQueryBuilder.java

```diff
@@ -292,15 +301,14 @@ protected QueryBuilder doRewrite(QueryRewriteContext queryRewriteContext) {
             if (vectorSupplier().get() == null) {
                 return this;
             }
-            KNNQueryBuilder knnQueryBuilder = new KNNQueryBuilder(fieldName(), vectorSupplier.get()).filter(filter());
-            if (maxDistance != null) {
-                knnQueryBuilder.maxDistance(maxDistance);
-            } else if (minScore != null) {
-                knnQueryBuilder.minScore(minScore);
-            } else {
-                knnQueryBuilder.k(k);
-            }
-            return knnQueryBuilder;
+            return KNNQueryBuilder.builder()
```

Comment:
please add comment on why we can set all maxDistance, minScore and k even if some of them can be null, if that was enabled in knn recently please add PR ref

Reply:
Yes the builder will throw an exception if any if the parameters or their combinations are incorrect. This querybuilder shouldn't be responsible for k-NN validations. Here is the PR https://github.com/opensearch-project/k-NN/pull/1707